tests/vectors_ed25519.rs

```diff
@@ -0,0 +1,67 @@
+// https://github.com/FiloSottile/mostly-harmless/blob/30a2d72/ed25519vectors/ed25519vectors.json
+
+#[cfg(test)]
+mod tests {
+
+    extern crate hex;
+
+    use hex::FromHex;
+
+    use std::fs;
+    use core::convert::TryFrom;
+    use salty::{PublicKey, Signature};
+    use salty::constants::{PUBLICKEY_SERIALIZED_LENGTH, SIGNATURE_SERIALIZED_LENGTH};
+
+    use serde::{Deserialize};
+    // use serde_json::Result;
+
+    #[derive(Deserialize)]
+    struct TestVector {
+        #[serde(rename = "A")]
+        a: String,
+        #[serde(rename = "R")]
+        r: String,
+        #[serde(rename = "S")]
+        s: String,
+        #[serde(rename = "M")]
+        msg: String,
+        #[serde(rename = "Flags")]
+        flags: Option<Vec<String>>,
+    }
+
+    #[test]
+    fn ed25519_test_case() {
+
+
+        let file = fs::File::open("tests/vectors/ed25519.json")
+            .expect("file should open read only");
+        let data: Vec<TestVector> = serde_json::from_reader(file)
+            .expect("file should be proper JSON");
+
+        for vector in data.iter() {
+            let pk_bytes = <[u8; PUBLICKEY_SERIALIZED_LENGTH]>::from_hex(&vector.a).expect("A should be in hex");
+            let pk = PublicKey::try_from(&pk_bytes);
+
+            const thirtytwo: usize = SIGNATURE_SERIALIZED_LENGTH / 2;
+            let r = <[u8; thirtytwo]>::from_hex(&vector.r).expect("R should be in hex");
+            let s = <[u8; thirtytwo]>::from_hex(&vector.s).expect("S should be in hex");
+            let mut signature_bytes: [u8; SIGNATURE_SERIALIZED_LENGTH] = [0; SIGNATURE_SERIALIZED_LENGTH];
+            signature_bytes[..thirtytwo].copy_from_slice(&r);
+            signature_bytes[thirtytwo..].copy_from_slice(&s);
+            let sig = Signature::from(&signature_bytes);
+
+            let result = pk.unwrap().verify(&vector.msg.as_bytes(), &sig);
+            if result.is_ok() {
+                if vector.flags.is_none() {
```

Comment:
If I understood correctly, flags tell us why it should fail. So flags = null => should verify the signature.
All other cases should fail. (I'm checking with the author)

If this is the case, currently we're getting a lot of error, likely because we don't check low order stuff.